models/remoteworkitem_test.go

```diff
@@ -0,0 +1,116 @@
+package models
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"testing"
+)
+
+func provideRemoteGithubData() ([]byte, error) {
+	url := "https://api.github.com/repos/almighty/almighty-core/issues/131"
+	response, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+
+	defer response.Body.Close()
+	responseData, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	return responseData, nil
+}
+
+func TestWorkItemMapping(t *testing.T) {
+
+	workItemMap := WorkItemMap{
+		AttributeExpression("title"): "system.title",
+	}
+	remoteWorkItem := RemoteWorkItem{ID: "xyz", Content: []byte("{\"title\":\"abc\"}")}
+
+	gh, err := NewGitHubRemoteWorkItem(remoteWorkItem)
```

Comment:
A little registry that maps ProviderGithub to the function NewGitHubRemoteWorkItem for lookup by the 'caller'